org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/gradle/GradleProgressAttributes.java

```diff
@@ -0,0 +1,178 @@
+import java.io.OutputStream;
+import java.util.List;
+
+import org.gradle.tooling.CancellationToken;
+import org.gradle.tooling.CancellationTokenSource;
+import org.gradle.tooling.ProgressListener;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.console.ProcessDescription;
+import org.eclipse.buildship.core.console.ProcessStreams;
+import org.eclipse.buildship.core.console.ProcessStreamsProvider;
+import org.eclipse.buildship.core.util.progress.CancellationForwardingListener;
+import org.eclipse.buildship.core.util.progress.DelegatingProgressListener;
+
+/**
+ * Holds attributes that are commonly used to handle progress in each Gradle invocation.
+ *
+ * @author Donat Csikos
+ */
+public final class GradleProgressAttributes {
+
+    private final ProcessStreams streams;
+    private final CancellationToken cancellationToken;
+    private final ImmutableList<ProgressListener> progressListeners;
+    private final ImmutableList<org.gradle.tooling.events.ProgressListener> progressEventListeners;
+
+    private GradleProgressAttributes(ProcessStreams streams, CancellationToken cancellationToken, List<ProgressListener> progressListeners,
+            List<org.gradle.tooling.events.ProgressListener> progressEventListeners) {
+        this.streams = Preconditions.checkNotNull(streams);
+        this.cancellationToken = Preconditions.checkNotNull(cancellationToken);
+        this.progressListeners = ImmutableList.copyOf(progressListeners);
+        this.progressEventListeners = ImmutableList.copyOf(progressEventListeners);
+    }
+
+    /**
+     * Returns the default output stream.
+     *
+     * @return the default output stream
+     */
+    public OutputStream getOutput() {
+        return this.streams.getOutput();
+    }
+
+    /**
+     * Returns the error stream.
+     *
+     * @return the error stream
+     */
+    public OutputStream getError() {
+        return this.streams.getError();
+    }
+
+    /**
+     * Returns the input stream.
+     *
+     * @return the input stream
+     */
+    public InputStream getInput() {
+        return this.streams.getInput();
+    }
+
+    /**
+     * Closes all input and output streams.
+     */
+    public void close() {
+        this.streams.close();
+    }
+
+    public List<ProgressListener> getProgressListeners() {
+        return this.progressListeners;
+    }
+
+    public List<org.gradle.tooling.events.ProgressListener> getProgressEventListeners() {
+        return this.progressEventListeners;
+    }
+
+    /**
+     * Returns the cancellation token associated to this instance.
+     *
+     * @return the cancellation token
+     */
+    public CancellationToken getCancellationToken() {
```

Comment:
⭕️ Do we really need getters here? Shouldn't there be a single method to apply the settings to a `ConfigurableLauncher`?

Reply:
That totally makes sense. I've just added the change.